agr_mvs/turno.py

```diff
@@ -0,0 +1,118 @@
+from datetime import datetime
+from agr_mvs.ticket import Ticket
+import math
+
+# Constante que nos expresa la longitud del turno que es 5'5 horas.
+HORAS_TURNO = 5
+
+class Turno:
+    horas_turno: list            # Mediante este array introducimos dos datetime que nos indican el inicio y el fin del turno
+    tickets: list                # tickets dentro de un Turno.
+    fecha:  datetime             # fecha del Turno           
+            
+    def __init__(self, horas_turno, S, fecha):
+        self.horas_turno = horas_turno
+        self.fecha = fecha
+        self.tickets = []
+        self.S = S
+
+
+    # Añadimos un Ticket a la lista de tickets del Turno
+    def addTicket(self, ticket):
+        if (self.fecha.date() == ticket.FechaTicket.date()):
+                if (ticket.FechaTicket.timestamp() >= self.horas_turno[0].timestamp() and ticket.FechaTicket.timestamp() <= self.horas_turno[1].timestamp()):
+                    self.tickets.append(ticket)
+        else:
+            # Excepción del tipo IndexError en caso de que no se pueda añadir el Ticket
+            raise IndexError("El Ticket no pertenece al Turno")
+        
+    '''
+    Para el cálculo del µ, se ha decidido que el valor de µ será el máximo de clientes que se han atendido en una hora.
+    Se va a dividir entre 60 para trabajar con minutos en los posteriores cálculos.
+    '''
+    def calculo_maximo_clientes_atendidos(self):
+        max_afluencia = 0
+        afluencia = 0
+        hora_actual = int(self.horas_turno[0].hour)
+        hora_fin = int(self.horas_turno[1].hour)
+        while (hora_actual <= hora_fin):
+            for ticket in self.tickets:
+                hora = ticket.FechaTicket.hour
+                if (hora == hora_actual):
+                    afluencia += 1
+            if (afluencia > max_afluencia):
+                max_afluencia = afluencia
+            hora_actual += 1
+            afluencia = 0
+        return (max_afluencia/60.0)
```

Comment:
TBH, no entiendo esta operación. Si quieres la máxima afluencia por minuto, no es la máxima afluencia por hora dividido por al número de minutos en una hora.

> Y no lo entiendo porque estás usando un literal en vez de una constante con nombre. Si esto se llamara MINUTOS_EN_HORA o como fuera, sería má sencillo de entender.

Reply:
Mediante esa operación se obtiene el numero máximo de afluencia que hemos tenido, una vez que sacamos el numero máximo de personas que hemos tenido en una hora ( dentro de la funcion nos vamos moviendo de hora en hora) lo dividimos entre 60 minutos que tiene una hora para expresarlo en minutos.

Reply:
¿Entiendes que la afluencia máxima por hora dividida por 60 **no es** el minuto con afluencia máxima? ¿No te das cuenta que esa afluencia máxima por minuto puede haber sucedido en la mitad o en el 25% de los minutos, dependiendo de la distribución? De veras, que hace tiempo que disteis estadística, pero si no sabéis que estáis haciendo, no lo hagáis...